src/picturepark-sdk-v1-angular/src/components/content-browser/content-browser.component.html

```diff
@@ -1,19 +1,116 @@
-                  [items]="items" 
-                  (update)="viewPortItems = $event" 
-                  (change)="onListChange($event)" 
-                  [style.height]="height"
-                  style="display:block">
-    <pp-content-browser-item *ngFor="let itemModel of viewPortItems"
-                             style="display: inline-block"
-                             [style.cursor]="selectionMode !== 'none' ? 'pointer' : ''"
-                             [itemModel]="itemModel" 
-                             [style.width]="(100 / columns) + '%'">
-    </pp-content-browser-item>
-  </virtual-scroll>
-  <div *ngIf="items.length === 0 && !isLoading">
-    {{'ContentBrowser.TextNoItems' | pptranslate}}
+<div class="content-browser" style="position: relative;" cdkScrollable>
+  <!-- TODO: translations, translate tooltips -->
+  <!-- TODO: tooltips for everything. -->
+  <!-- TODO: move styles -->
+  <div class="content-browser__header">
+    <mat-toolbar>
+      <div style="display: flex; width: 100%;">
+        <button mat-button [matMenuTriggerFor]="selectMenu"> Select
+          <mat-icon>more_vert</mat-icon>
+        </button>
+
+        <mat-menu #selectMenu="matMenu">
+          <button mat-menu-item (click)="toggleItems(true);">Select first {{ items.length }}</button>
+          <button mat-menu-item (click)="toggleItems(false);">Deselect all</button>
+        </mat-menu>
+
+        <mat-divider [style.margin]="'0 8px'" [vertical]="true"></mat-divider>
+
+        <button *ngIf="selectedItems.length === 1" mat-icon-button [matTooltip]="'Preview'">
+          <mat-icon aria-label="Preview item" (click)="previewItem(selectedItems[0])">open_in_new</mat-icon>
+        </button>
+        <button *ngIf="selectedItems.length > 0" mat-icon-button (click)="downloadItems()" [matTooltip]="'Download'">
+          <mat-icon aria-label="Download selected items">get_app</mat-icon>
+        </button>
+
+        <div style="flex-grow: 1;"></div>
+        <span *ngIf="totalResults !== null" style="align-self: center" class="mat-body-1"> {{totalResults}} contents </span>
+
+        <button mat-icon-button [matMenuTriggerFor]="sortMenu" [matTooltip]="'Sort'">
+          <mat-icon aria-label="Select sorting">sort</mat-icon>
+        </button>
+
+        <mat-menu #sortMenu="matMenu">
+          <button mat-menu-item [disabled]="activeSortingType === sortingTypes.relevance" (click)="isAscending = true; update();">
+            <mat-icon *ngIf="isAscending === true">check</mat-icon>
+            <span>Ascending</span>
+          </button>
+          <button mat-menu-item [disabled]="activeSortingType === sortingTypes.relevance" (click)="isAscending = false; update();">
+            <mat-icon *ngIf="isAscending === false">check</mat-icon>
+            <span>Descending</span>
+          </button>
+          <mat-divider></mat-divider>
+          <button mat-menu-item (click)="setSortingType(sortingTypes.relevance)">
```

Comment:
Move the items to the Component and create an array of sorting items.

Reply:
Done